detectors/gcp/detector.go

```diff
@@ -46,10 +46,10 @@ func (d *Detector) CloudPlatform() Platform {
 	switch {
 	case d.onGKE():
 		return GKE
-	case d.onCloudFunctions():
-		return CloudFunctions
 	case d.onCloudRun():
 		return CloudRun
+	case d.onCloudFunctions():
+		return CloudFunctions
```

Comment:
Ah I think this needs to have cloud functions first, see https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/552:

> The detector was incorrectly detecting the cloud run platform instead of the cloud functions platform. This is because cloud functions v2 is implemented on top of cloud run. This commit makes the detector check for cloud functions first to fix this issue.

We should at least leave a comment here about why this ordering is important. It would be better if we can find a way to more soundly fix this rather than relying on the order, but I'm not sure that will be possible if cloud functions is built on cloud run.

Reply:
I remembered. will fix

Reply:
Cloud Functions gen2 also reserved the `K_CONFIGURATION` environment variable, but undocumented... :(
Well, therefore wrong detection if change the `switch...case` order.

As you said, we should leave a details comment here.